### PR TITLE
Ports code cleanup from spacepods.

### PR DESCRIPTION
**Pull request by ghost**
Yeah, this ports the new obj return_temperature from spacepod (removing the snowflake code from living.dm), and also slightly better multi tile support!

### CODE REVIEW

File: code/_onclick/adjacent.dm [hunk 1]
```diff
@@ -72,6 +72,11 @@
 		return FALSE
 	if(loc.Adjacent(neighbor,target = neighbor, mover = src))
 		return TRUE
+	for(var/turf/T in locs) //this is to handle multi tile objects
+		if(isnull(T))
```
Comment (RemieRichards) [inline code/_onclick/adjacent.dm]: you can't have nulls in your locs.

Comment (MoreRobustThanYou) [inline code/_onclick/adjacent.dm]: Better be safe than sorry, I guess?

Comment (Cyberboss) [inline code/_onclick/adjacent.dm]: no, it's literally impossible

File: code/_onclick/adjacent.dm [hunk 2]
```diff
@@ -72,6 +72,9 @@
 		return FALSE
 	if(loc.Adjacent(neighbor,target = neighbor, mover = src))
 		return TRUE
+	for(var/turf/T in locs) //this is to handle multi tile objects
```
Comment (Cyberboss) [inline code/_onclick/adjacent.dm]: This should actually be a typeless for

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/_onclick/adjacent.dm (hunk 2 above)
Comment (MoreRobustThanYou) [inline code/_onclick/adjacent.dm]: yeah, well changing it to var/T causes errors, sooo

Comment (AnturK) [inline code/_onclick/adjacent.dm]: Does the likely expensive check twice for negative results for every single tile objects now.
Does not handle  checks against multi-tile objects only from.
Does not change 9999 uses of get turf as the adjacency check in the current code.
There's more issues with this but these are the major ones.

**Review by AnturK — CHANGES_REQUESTED** [review]
Why are you trying to sneak in change that explicitly needs a bigger refactor to work properly when i told you this repeatedly.

File: code/_onclick/adjacent.dm [hunk 3]
```diff
@@ -72,6 +72,10 @@
 		return FALSE
 	if(loc.Adjacent(neighbor,target = neighbor, mover = src))
 		return TRUE
+	if(multitile)
```
Comment (RemieRichards) [inline code/_onclick/adjacent.dm]: no.

```if(bounds_width != world.icon_size || bounds_height != world.icon_size)```

**Pull request closed without merging**